lab_2/backend/src/main/scala/infrastructure/aws/service/DynamoDbUserService.scala

```diff
@@ -0,0 +1,66 @@
+package infrastructure.aws.service
+
+import cats.Applicative
+import cats.effect.kernel.Sync
+import cats.syntax.all._
+import domain.model.{Credentials, User}
+import domain.spi._
+import org.scanamo.generic.auto._
+import org.scanamo.syntax._
+import org.scanamo.{DynamoReadError, Scanamo, Table}
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+
+import java.util.UUID
+
+class DynamoDbUserService[F[_] : Sync](dynamoDbClient: DynamoDbClient) extends UserService[F] {
+
+  private lazy val dbExecutor = Scanamo(dynamoDbClient)
+  private lazy val usersTable = Table[User]("Users")
+
+  override def createUser(user: User): F[Either[UserCreationError, Unit]] = for {
+    queryRes <- Sync[F].blocking(dbExecutor.exec(usersTable.scan()))
+    users <- queryRes.collect {
+      case Right(user) => Applicative[F].pure(user)
+    }.sequence
+    putRes <- if (users.exists(_.login.equals(user.login))) {
+      Applicative[F].pure(Either.left(UserCreationError("User with given login already exists.")))
```

Comment:
I am not sure if this message has proper content because by saying that a user with this `login` exists, you give an insight into our data

Reply:
What do you propose then?

Reply:
We discussed that we should respond with generic content, like "Invalid data provided". But internally, such a message could be useful.